code/game/machinery/doors/firedoor.dm

```diff
@@ -160,6 +313,8 @@
 
 	if(density)
 		open()
+		if(alarm_type)
+			addtimer(CALLBACK(src, .proc/correct_state), 2 SECONDS, TIMER_UNIQUE)
```

Comment:
Why this two second timer?

Reply:
This is what makes a firelock close if it was crowbar'd open while active now. Rather than the fire alarm essentially pulsing all firelocks in the area to close every set number of seconds, a firelock will now close exactly two seconds after being opened.

The `correct_state()` proc will observe the `alarm_type` var, and will refuse to close if the alarm state has cleared within those two seconds.

(This is when the firelock is just opened quickly, not when it's being held open. There's a separate block of code to handle people walking away after holding a firelock open.)